contrib/mesos/pkg/scheduler/plugin_test.go

```diff
@@ -200,44 +201,56 @@ func (lw *MockPodsListWatch) Pod(name string) *api.Pod {
 	return nil
 }
 func (lw *MockPodsListWatch) Add(pod *api.Pod, notify bool) {
-	lw.lock.Lock()
-	defer lw.lock.Unlock()
+	func() {
+		lw.lock.Lock()
+		defer lw.lock.Unlock()
+		lw.list.Items = append(lw.list.Items, *pod)
+	}()
```

Comment:
I think it's unfortunate that we work around implementation details of the fakeWatcher inside MockPodsListWatch. By looking at the code it is not clear why we need to unlock prematurely.